modules/fundamental/src/sbom/model/details.rs

```diff
@@ -255,11 +259,31 @@ impl SbomAdvisory {
 
             let mut packages = vec![];
             if let Some(package) = &product.product_status.package {
-                let package = SbomPackage {
-                    name: package.to_string(),
-                    ..Default::default()
+                // We need to enable search on namespace/name pattern as well
+                // Also, from the brief look at it, the search is case sensitive
+                // which doesn't cover all use cases
+                let parts = package.split('/').collect::<Vec<_>>();
+                let name = if parts.len() == 1 { parts[0] } else { parts[1] };
+
+                let query = Query {
+                    q: format!("name={}", name),
+                    sort: Default::default(),
                 };
-                packages.push(package);
+
+                let mut sbom_packages = sbom_service
+                    .fetch_sbom_packages(summary.head.id, query, Default::default(), ())
```

Comment:
I think this should pass on the transaction.

Reply:
I didn't find any examples where we are getting `Transactional` from `ConnectionOrTransaction`. I tried quickly implementing something like

```
impl<'db> From<ConnectionOrTransaction<'db>> for Transactional {
    fn from(conn_or_tx: ConnectionOrTransaction<'db>) -> Self {
        match conn_or_tx {
            ConnectionOrTransaction::Connection(_) => Transactional::None,
            ConnectionOrTransaction::Transaction(tx) => Transactional::Some((*tx).clone()),
        }
    }
}
```
But failed miserably as `DbTransaction` is not cloneable and trying to make some changes to `Transactional` cascades all over the place.
If you have some example on how to do this easily, please let me know. Otherwise, I would deal with it in a separate PR as it is potentially affecting a lot of code.

Reply:
Ok, I just peeked into this. `ConnectionOrTransaction` carries exactly that. Where the other function accepts an indication if a new connection should be created, or an existing transaction be re-used. Those two concepts conflict with each other.

Normally, the more "outer" service functions accept the `Transactional`, getting a new database from the connection pool is required. While the more inner ones accept a an existing "connection or transaction". Especially when we're talking about the "from model" functions, which don't have a connection pool themselves.

Honestly, it feels a bit like we're mixing things here. My concern also is that this will be a performance issue.

I guess the only feasible option is to have `fetch_sbom_packages`  accept a `ConnectionTrait` instead of an `AsRef<Transactional>`. Maybe create a `fetch_sbom_packages_with` and have it accept the `ConnectionTrait`, keeping the existing one with the `AsRef<Transactional>`.

Reply:
Sounds good, thanks. Yeah, I think the new function is needed here anyways as the current one doesn't do all functionalities needed. And we need to provide a way to have in-mem graph implementation as well.